File: JavaScript/selection_sort.js
let selectionSort = (arr) => {
    let length = arr.length;
    for (let i = 0; i < length; i++) {
        let min = i;
        for (let j = i + 1; j < length; j++) {
            if (arr[min] > arr[j]) {
                min = j;
            }
        }
        if (min !== i) {
            // Swapping the elements
            let tmp = arr[i];
            arr[i] = arr[min];
            arr[min] = tmp;
        }
    }
    return arr;
}

let arr = [10,12,9,25,3];
selectionSort(arr);
console.log(arr);

// Output: [3, 9, 10, 12, 25]

File: Julia/selectionSort.jl
#Selection Sort Function
function selectionsort!(arr)        #The function changes its argument hence bang(!) is used. 
    for i in 1:length(arr)          #Looping through array
        
        for j in i+1:length(arr)
            
            if arr[i] > arr[j]      #Swapping elements if next element is greater than current element
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp
            end
        
        end
    
    end
    
    return arr                      #returning the sorted array
end

#Driver code

A = [4,3,5,1,2]         #input
println(A)              #printing array before sorting
selectionsort!(A)        #calling sorting function
println(A)              #printing array after sorting